db/transaction.go

```diff
@@ -204,6 +204,8 @@ func (tx *Transaction) Commit() (ref types.Ref, err error) {
 	defer tx.db.lock()()
 	_, err = tx.db.noms.FastForward(tx.db.noms.GetDataset(LOCAL_DATASET), ref)
 	if err != nil {
```

Comment:
The documentation for `FastForward` is not great about what errors it can return, but looking through the code, I believe it returns specifically `ErrMergeNeeded` in this case, and never returns any other type of error.

It would be good to log an error if we see any other type of error for self-documentation purposes.

Reply:
Good idea. I'll look at noms code too and maybe just reuse that error

Reply:
I ended up keeping CommitError. I felt that it was cleaner to keep this logic out of api.go